### PR TITLE
CORTX-33805 - Added local registry support

### PR DESCRIPTION
**Pull request by shailesh-vaidya**
# Problem Statement
- CORTX deployment was failing as locally generated images were not tagged. Added support for setting up a local registry and pushing images into it. This will also help in comunity multinode deployment support. 

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM
- [x] Jenkins pipelines used for testing - https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/Community/job/Community-deploy/52/console

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [x] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: docker/cortx-deploy/build.sh [hunk 1]
```diff
@@ -97,6 +94,13 @@ echo cortx-rgw:"$(awk -F['-'] '/'ceph-base'/{ print $5  }' RELEASE.INFO | cut -d
 echo cortx-re:"$(git rev-parse --short HEAD)",
 }
 
+function setup_local_image_registry {
+docker rm -f local-registry
```
Comment (gauravchaudhari02) [inline docker/cortx-deploy/build.sh:98]: can we handle failure of commands in this function with error message? and please align code in function properly

File: docker/cortx-deploy/build.sh [hunk 2]
```diff
@@ -119,12 +123,19 @@ CREATED_DATE=$(date -u +'%Y-%m-%d %H:%M:%S%:z')
 
 docker-compose -f ./docker-compose.yml build --parallel --force-rm --compress --build-arg GIT_HASH="$GIT_HASH" --build-arg VERSION="$VERSION-$DOCKER_BUILD_BUILD" --build-arg CREATED_DATE="$CREATED_DATE" --build-arg BUILD_URL=$BUILD_URL --build-arg ENVIRONMENT=$ENVIRONMENT --build-arg OS=$OS --build-arg OS_TYPE=$OS_TYPE --build-arg OS_RELEASE=$OS_RELEASE --build-arg CORTX_VERSION="$CORTX_VERSION" $SERVICE
 
+
+if [ "$REGISTRY" == "local" ]; then
+echo "Setting up local container image registry"
```
Comment (gauravchaudhari02) [inline docker/cortx-deploy/build.sh:128]: please align code correctly

File: docker/cortx-deploy/build.sh [hunk 3]
```diff
@@ -119,12 +123,19 @@ CREATED_DATE=$(date -u +'%Y-%m-%d %H:%M:%S%:z')
...
+REGISTRY="$HOSTNAME:8080"
+fi
+
 for SERVICE_NAME in $SERVICE
 do
 if [ "$DOCKER_PUSH" == "yes" ];then
         echo "Pushing Docker image to GitHub Container Registry"
         docker tag $SERVICE_NAME:$TAG $REGISTRY/$PROJECT/$SERVICE_NAME:$TAG
-        docker push $REGISTRY/$PROJECT/$SERVICE_NAME:$TAG
+#        docker push $REGISTRY/$PROJECT/$SERVICE_NAME:$TAG
```
Comment (gauravchaudhari02) [inline docker/cortx-deploy/build.sh:138]: why we commented this line?

File: solutions/community-deploy/build-cortx.sh [hunk 4]
```diff
@@ -129,8 +130,9 @@ function nginx_validation() {
 nginx_validation
 # clone cortx-re repository & run build.sh
 if [ -d $LOCAL_MOUNT/cortx-re ]; then rm -rf $LOCAL_MOUNT/cortx-re; fi
-git clone https://github.com/Seagate/cortx-re -b $BRANCH && pushd cortx-re/docker/cortx-deploy/
-./build.sh -b http://$IP -o rockylinux-8.4 -s all -e opensource-ci
+#git clone https://github.com/Seagate/cortx-re -b $BRANCH && pushd cortx-re/docker/cortx-deploy/
+git clone https://github.com/shailesh-vaidya/cortx-re -b community-deploy && pushd cortx-re/docker/cortx-deploy/
```
Comment (gauravchaudhari02) [inline solutions/community-deploy/build-cortx.sh:134]: please remove this once testing is done

File: solutions/community-deploy/build-cortx.sh [hunk 5]
```diff
@@ -129,8 +130,9 @@ function nginx_validation() {
...
+./build.sh -b http://$IP -o rockylinux-8.4 -s all -e opensource-ci -r local -p yes
```
Comment (gauravchaudhari02) [inline solutions/community-deploy/build-cortx.sh:135]: need to add variable for registry option?

File: jenkins/deployment/community/cortx-community-deploy.groovy [hunk 6]
```diff
@@ -70,99 +73,103 @@ pipeline {
             '''
             }
         }
+
         stage ('Network and storage configuration') {
             steps {
                 script { build_stage = env.STAGE_NAME }
+                script {
+                    env.AWS_IP = sh( script: '''
+                    cd solutions/community-deploy/cloud/AWS && terraform show -json terraform.tfstate | jq .values.outputs.cortx_deploy_ip_addr.value | tr -d '""'
```
Comment (gauravchaudhari02) [inline jenkins/deployment/community/cortx-community-deploy.groovy:82]: `tr -d '"'` won't work here?

File: docker/cortx-deploy/build.sh (hunk 2 above)
Comment (shailesh-vaidya) [reply docker/cortx-deploy/build.sh]: Fixed

File: docker/cortx-deploy/build.sh (hunk 3 above)
Comment (shailesh-vaidya) [reply docker/cortx-deploy/build.sh]: Fixed

File: solutions/community-deploy/build-cortx.sh (hunk 5 above)
Comment (shailesh-vaidya) [reply solutions/community-deploy/build-cortx.sh:135]: It's just a `local` string and then the script will set up a local registry.

File: jenkins/deployment/community/cortx-community-deploy.groovy (hunk 6 above)
Comment (shailesh-vaidya) [reply jenkins/deployment/community/cortx-community-deploy.groovy]: Fixed

File: docker/cortx-deploy/build.sh (hunk 1 above)
Comment (shailesh-vaidya) [reply docker/cortx-deploy/build.sh]: Fixed.

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request merged**